Cclave 2207/airlock new UI

FabianDEV08 opened the pull request:
## Overview

The following PR contains changes to update the UI if the web wormhole app.

## Current Behavior

The app has the UI of the original web wormhole app.

## Expected Behavior

The app should update the UI to accordingly to the figma designs.

## Changes

- Added required css styles.
- Updated index.html according with the new UI.
- Updated main.js, removed unnecesary variables that were linked to deleted html elements, not needed in the new UI.

## Resources 

<img width="384" alt="Screen Shot 2021-11-19 at 1 43 05 PM" src="https://user-images.githubusercontent.com/26255205/142682425-535f1c94-8d25-465c-86b0-f68152f64dc1.png">
<img width="381" alt="Screen Shot 2021-11-19 at 1 43 31 PM" src="https://user-images.githubusercontent.com/26255205/142682442-cff99167-75d5-4230-adce-2a6fd7371a02.png">

FabianDEV08 committed:
Update index.html

Created elements for the new UI of the Airlock webapp

FabianDEV08 committed:
Update style.css

Added necessary styles and colors for the Airlock webapp.

FabianDEV08 committed:
Added Assets

Added assets for the new UI look and feel

FabianDEV08 committed:
Various changes

- Assest relocation
- main.js updated to use the new UI elements

FabianDEV08 committed:
Multiple Changes

- Added svg for default file image.
- Updated index.html to add the progress bar for file download, removed hardcoded class, added script for converting bytes into KB, MB, GB etc.
- Updated styles according to the different body states, created style for progress, added media query for devices with screen width lower than 340px.
- Added format-file-size.js for including the correct byte weight in the download card.
- Updated main.js to remove lines of code that were not supposed to be called during the download, removed code that populated the file card with the webwormhole UI, removed interaction with the file card, added default img for the card file icon.

FabianDEV08 committed:
Removed autocomplete

Removed autocomplete element from index.html, removed autocomplete references in download flow.

FabianDEV08 committed:
Removed overflow in #main

Removed property that was causing the airlock mobile view to display a gray thin bar at the bottom.

FabianDEV08 committed:
Various changes

- Removed unnecesary validation in displayInputLabel function.
- Added right margin to fileName label.

franc5 commented:
Please, do not include GraphikRegular.otf

franc5 commented:
Please, include the license for ~~GraphikRegular.otf~~ OverpassMono-Bold.ttf (**EDITED**)

franc5 commented on web/style.css:
```diff
@@ -23,6 +28,16 @@
+@font-face {
+font-family: "Graphik";
+src: url(assets/fonts/GraphikRegular.otf);
+}
```
Please, remove these lines. See also https://github.com/SpiderOak/webwormhole/pull/7#issuecomment-975576561

FabianDEV08 replied:
The font has been removed.

franc5 commented:
If the user enters an invalid phrase, there's no way to go back an enter a new one. Let's add a "Enter another phrase" button to allow our users to do so.

Also, please note that this should be "Not a valid Airlock phrase."

<img width="340" alt="Screenshot 2021-11-22 at 11 27 40" src="https://user-images.githubusercontent.com/64472239/142879155-848307dc-56ab-4cd5-8e7f-765c00cd23ab.png">

franc5 commented:
Once the file is successfully downloaded, we show a "Disconnected." message. Let's change it to something like "File successfully downloaded". Also, let's add an "Enter another phrase" button (the same button suggested in https://github.com/SpiderOak/webwormhole/pull/7#issuecomment-975586107) to allow our users to enter a new phrase.

franc5 commented:
If the user enters an expired ~~wormhole~~ airlock phrase, the following message is shown "No such slot. The wormhole might have expired.". I guess this should be "No such slot. The phrase might have expired.", but let's ask Zach about the correct wording in these case.

Similar to added comments, please add an "Enter another phrase" button.

franc5 commented:
During the "Connecting..." phase, we don't show the filename nor the file size in the file card. Don't we have these values? If so, let's write something like "Loading file..." instead of its name. Please, validate this with Zach before continuing.

<img width="435" alt="Screenshot 2021-11-22 at 11 36 21" src="https://user-images.githubusercontent.com/64472239/142880496-08c07891-daa1-405f-a178-9ba6965c6218.png">

FabianDEV08 commented:
> No such slot. The phrase might have expired.

when connecting the file download is not triggered instantly, when the data for the file is obtanied then the name and the size appear, I can add the "Loading file..." text to the file card to make more sense.

brownbike commented:
Small style thing with the Get File button. You have some default styles causing a split border color.
<img width="165" alt="Screen Shot 2021-11-22 at 10 08 54 AM" src="https://user-images.githubusercontent.com/2212315/142913335-2314c248-f2a1-4699-8b3c-d1019f5374ea.png">
Looking at the figma, we should have a border.

FabianDEV08 committed:
Deleted font

Deleted unused font.

brownbike reviewed (approved):
Looks good, I tested this on MacOS

FabianDEV08 committed:
updated info messages

Updated a few strings from the info html element to avoid using wormhole text.

FabianDEV08 committed:
Vaious UI changes

- Created retry button to reset the receiving process.
- Created a card Skeleton for displaying while waiting for the file download.

FabianDEV08 committed:
Updated retry

Updated retry function to correctly reload the html to go back to insert a new phrase.

FabianDEV08 commented:
> If the user enters an expired ~wormhole~ airlock phrase, the following message is shown "No such slot. The wormhole might have expired.". I guess this should be "No such slot. The phrase might have expired.", but let's ask Zach about the correct wording in these case.
> 
> Similar to added comments, please add an "Enter another phrase" button.



> Once the file is successfully downloaded, we show a "Disconnected." message. Let's change it to something like "File successfully downloaded". Also, let's add an "Enter another phrase" button (the same button suggested in [#7 (comment)](https://github.com/SpiderOak/webwormhole/pull/7#issuecomment-975586107)) to allow our users to enter a new phrase.

FabianDEV08 commented:
Closed it by mistake, this PR is still active.

FabianDEV08 committed:
Update style.css

Removed borderline from the "Get file" button.

FabianDEV08 committed:
Removed unused font

FabianDEV08 commented:
> Please, do not include GraphikRegular.otf

The font has been removed.

FabianDEV08 committed:
Added Font License

Added Overpass Mono License and disclaimer for the airlock web app.

FabianDEV08 commented:
> Please, include the license for ~GraphikRegular.otf~ OverpassMono-Bold.ttf (**EDITED**)

I've just added the files for the license, please check it out and let me know if it's ok, it's my first time adding a font license and I copied what we have in the share-ui repo.

FabianDEV08 committed:
Updated OverpassMono License

Updated the OFL.txt, removed the disclaimer.sh file, removed the declaration of the script in package.json.

FabianDEV08 commented:
> Small style thing with the Get File button. You have some default styles causing a split border color. <img alt="Screen Shot 2021-11-22 at 10 08 54 AM" width="165" src="https://user-images.githubusercontent.com/2212315/142913335-2314c248-f2a1-4699-8b3c-d1019f5374ea.png"> Looking at the figma, we should have a border.

I've updated the style to get rid of the weird border.

franc5 commented:
The site looks good. Approving this PR

franc5 reviewed (approved):
The
